src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/ProjectSystem/RazorProjectHostBase.cs

```diff
@@ -157,11 +166,7 @@ private ProjectSnapshotManagerBase GetProjectManager()
             return _projectManager;
         }
 
-        protected async Task UpdateAsync(Action action)
-        {
-            await CommonServices.ThreadingService.SwitchToUIThread();
-            action();
-        }
+        protected Task UpdateAsync(Action action) => _foregroundDispatcher.RunOnForegroundAsync(action, CancellationToken.None);
```

Comment:
The magic! That being said, should this take a cancellation token now? We didn't do it before which was probably just an oversight?